main.go

```diff
@@ -147,9 +148,21 @@ func printDiff(a, b string, metadata []jd.Metadata) {
 	}
 	diff := aNode.Diff(bNode, metadata...)
 	if *output == "" {
-		fmt.Print(diff.Render())
+		str = diff.Render()
+		if len(str) > 0 {
+			fmt.Print(str)
+			os.Exit(1)
+		} else {
+			os.Exit(0)
+		}
```

Comment:
A more idiomatic way to check for the string "zero" value is just to check for `""`.  And you don't need to declare the `str` var ahead of time.  And `os.Exit()` will terminate immediately, so there's no need for the `if else` structure. 
 Something like this will do:

```golang
str := diff.Render()
if str == "" {
	os.Exit(0)
}
fmt.Print(str)
os.Exit(1)
```

You can follow this pattern for all 4 locations and delete the `str` var at the top.

Reply:
Thanks, I've made the suggested changes.